### ISSUE TITLE
[proposed enhancement] Add optional C++20 module support

### ISSUE THREAD

**Issue by Theramar** [z4kn4fein/cpp-semver#7]
Instead of downloading `semver.hpp` users could install an optional `semver.cppm`

simple solution:
```cpp

//semver.cppm. This is an automatically generated file
#define SEMVER_MODULE
//copy "semver.hpp" content here

```

```cpp

//semver.hpp
#ifdef SEMVER_MODULE
#define SEMVER_EXPORT export
#else
#define SEMVER_EXPORT
#endif

SEMVER_EXPORT namespace semver { 
//...
}
```

**Comment by kelteseth** [comment]
We now track cpp-semver modules support progress at https://arewemodulesyet.org/ . Feel free to create a PR if the status changes at https://github.com/kelteseth/arewemodulesyet/ 😊

**Comment by z4kn4fein** [comment]
Hi @Theramar, I've released `v0.4.0`, which includes the support for C++ 20 modules. Could you please check that it's working as you expected? Thanks!

@kelteseth, thank you for letting me know! I'll create a PR to report the lib's current status.

**Comment by Theramar** [comment]
It works! Thank you
